pubsub/src/types.rs

```diff
@@ -62,6 +62,14 @@ impl From<String> for SubscriptionId {
 	}
 }
 
+// TODO: Don't unwrap, and probably implement TryFrom instead of From
+use std::convert::TryInto;
+impl From<usize> for SubscriptionId {
```

Comment:
I think instead of providing this `From<usize>` implementation we should rather change the `NumericIdProvider` to convert `usize` to `u64` internally. The reason is that it's highly unlikely for `NumericIdProvider` to reach the case where this conversion might fail (given it runs on 128-bit processor anyway O.o) and here we are making stronger assumption that `usize` can be safely converted to `SubscriptionId`.